submissions/LuckyDnepr/OOP-Exercise/index.js

```diff
@@ -0,0 +1,144 @@
+"use strict";
+
+class Inhabitant {
+  constructor(species, name) {
+    this.species = species;
+    this.name = name;
+  }
+  getInfo() {
+    return `I'm ${this.species} and my name is ${this.name}.`;
+  }
+}
+
+class Saying {
```

Comment:
We can teach different residents to speak. In addition, we may have inhabitants who cannot speak, such as the Goldfish. In addition, each inhabitant will have its own vocabulary.
Perhaps this skill can be made into a function, but with classes, it seems more obvious to me.

Reply:
Why cannot this be a part of class?
For e.g. not every inhabitant has hands or friends, yet we do not have dedicated classes for hands and/or friends.

Reply:
This can be made part of the class. But, since JS doesn't have multiple inheritance, in the case of Catwoman (physical data of Woman and text of Cat), I would have to use "mixin" to mix in a specific function.
Another option: you could make this class a static method of the parent class, and call it directly:
Inhabitant.saying(new Cat(), "vocabulary");

And about "friends" I had some options with implementation of property of object "friends" as object of class "SocialTies" with methods "get", "set" and "del".

Also, having limbs as a class doesn't make sense, since there's nowhere to extend it.

Reply:
Shared vocabulary is a valid use case. Static method would do.

Both approaches are valid. The truth is that composition in this cases works better that classic OOP. The goal of this exercise is to try it with OOP and realise OOP's flaws in these regards.

Reply:
Friends is a valid case, if we have requirement to have more processing rules like those you put in your example. Otherwise it might add unncessary complexity to the code. The more code we write the more efforts are required to maintain it.
This is to be considered when implementing solutions.

Reply:
I need to get more familiar with the composition. I've read several approaches to implementation, but I haven't ventured into it.